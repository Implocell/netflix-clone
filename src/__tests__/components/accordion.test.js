import { render, fireEvent, queryByText } from "@testing-library/react";
import faqsData from "../../fixtures/faqs.json";
import { Accordion } from "../../components";

describe("<Accordion />", () => {
	test("renders the <Accordion /> with populated data", () => {
		const { container, getByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{faqsData.map((item) => (
					<Accordion.Item key={item.id}>
						<Accordion.Header>{item.header}</Accordion.Header>
						<Accordion.Body>{item.body}</Accordion.Body>
					</Accordion.Item>
				))}
			</Accordion>
		);

		expect(getByText("Frequently Asked Questions")).toBeTruthy();
		expect(getByText("What is Netflix?")).toBeTruthy();
		expect(getByText("How much does Netflix cost?")).toBeTruthy();
		expect(getByText("Where can I watch?")).toBeTruthy();
		expect(getByText("How do I cancel?")).toBeTruthy();
		expect(getByText("What can I watch on Netflix?")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	test("should open and close the accordion", () => {
		const { container, queryByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{faqsData.map((item) => (
					<Accordion.Item key={item.id}>
						<Accordion.Header>{item.header}</Accordion.Header>
						<Accordion.Body>{item.body}</Accordion.Body>
					</Accordion.Item>
				))}
			</Accordion>
		);
		const whatIsNetFlixBodyText =
			"Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

		expect(queryByText(whatIsNetFlixBodyText)).toBeFalsy();
		fireEvent.click(queryByText("What is Netflix?"));
		expect(queryByText(whatIsNetFlixBodyText)).toBeTruthy();

		fireEvent.click(queryByText("What is Netflix?"));
		expect(queryByText(whatIsNetFlixBodyText)).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
