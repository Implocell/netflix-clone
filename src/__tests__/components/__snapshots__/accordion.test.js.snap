// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-l3iym9-0 fXsxPD"
>
  <div
    class="accordion__Inner-l3iym9-1 hhulRI"
  >
    <h1
      class="accordion__Title-l3iym9-5 dlhLjy"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        What is Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        How much does Netflix cost?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        Where can I watch?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        How do I cancel?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        What can I watch on Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> should open and close the accordion 1`] = `
<div
  class="accordion__Container-l3iym9-0 fXsxPD"
>
  <div
    class="accordion__Inner-l3iym9-1 hhulRI"
  >
    <h1
      class="accordion__Title-l3iym9-5 dlhLjy"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        What is Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        How much does Netflix cost?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        Where can I watch?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        How do I cancel?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-l3iym9-2 bgTtqM"
    >
      <div
        class="accordion__Header-l3iym9-3 ipyhCc"
      >
        What can I watch on Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
